### ISSUE TITLE
[Bug]: 'OneSignalFramework/OneSignalFramework.h' file not found 

### ISSUE THREAD

**Issue by fw-amgaa** [OneSignal/onesignal-expo-plugin#231]
### What happened?

Created clean expo project, followed this doc: https://documentation.onesignal.com/docs/react-native-expo-sdk-setup.
Then I built native ios to test it on my phone, but this error keeps occurring.

### Steps to reproduce?

```Markdown
#import <OneSignalFramework/OneSignalFramework.h>
^
'OneSignalFramework/OneSignalFramework.h' file not found
```

### What did you expect to happen?

I expect it to work properly when I build to native IOS from expo. 

### OneSignal Expo SDK version

2.0.2

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by mrrenaud** [comment]
@fw-amgaa Could you show you plugins config in app.json ?

**Comment by mikeislearning** [comment]
I'm having the same issue, my plugin config looks like:

```
plugins: [
...
  [
        "onesignal-expo-plugin",
        {
          mode: "development",
        },
      ],
```

**Comment by codeagus11** [comment]
Same issue here!

**Comment by codeagus11** [comment]
Please move onesignal-expo-plugin at the root and works fine

**Comment by louwjlabuschagne** [comment]
Also struggling with this, @codeagus11 what do you mean with `at the root`? Just first in the array? Or is there somewhere else we should put this?

**Comment by codeagus11** [comment]
Correct, you should put the `onesignal-expo-plugin` as a first element of the array.

EAS is not related to the app.json (or app.config.js). The configuration in this file will change the native properties (inside the android and ios folders).

**Comment by louwjlabuschagne** [comment]
@codeagus11 this solves the problem when building the app in the GUI, however, if we try and build using xcode CLI using 

```
xcodebuild -scheme "$SCEHEME" \
-archivePath $TMP/$SCEHEME.xcarchive \
-sdk iphoneos \
-configuration Release \
-destination generic/platform=iOS \
DEVELOPMENT_TEAM=$TEAM_ID
```

We still get the error - any ideas? It seems like the framework needs XCode GUI to ensure the linking to the files?

**Comment by codeagus11** [comment]
> @codeagus11 this solves the problem when building the app in the GUI, however, if we try and build using xcode CLI using
> 
> ```
> xcodebuild -scheme "$SCEHEME" \
> -archivePath $TMP/$SCEHEME.xcarchive \
> -sdk iphoneos \
> -configuration Release \
> -destination generic/platform=iOS \
> DEVELOPMENT_TEAM=$TEAM_ID
> ```
> 
> We still get the error - any ideas? It seems like the framework needs XCode GUI to ensure the linking to the files?

try with `npx expo prebuild --clean` before

**Comment by ChristopherGabba** [comment]
I can confirm. I'm still experiencing this issue. Moving the one-signal plugin to the first plugin fixed the problem per @mikeislearning 's solution

**Comment by ltvan** [comment]
Me too. The error occurs when building with xcodebuild. But if I build using GUI, it's fine. Also, expo run:ios is also fine. Not sure what's the problem is.

This fork mentioned the same issue, is it related to this issue?
https://github.com/eriksape/react-native-onesignal?tab=readme-ov-file#issue-3---symbols-not-found-for-architecture-x86_64-andor-onesignalonesignalh-file-not-found

**Comment by ltvan** [comment]
I known what happened. The Azure DevOps build task default to **/*.xcodeproj/project.xcworkspace. To fix the issue, we need to use ios/*.xcworkspace.

**Comment by fannytavart** [comment]
I am upgrading my app to Expo SDK 53, and am still facing this issue with onesignal-expo-plugin 2.4.0. I tried moving the plugin at the top of the plugin list but it didn't work for me. Has anyone found a way to fix this issue?

**Comment by fannytavart** [comment]
For anyone still getting this error, I got in touch with OneSignal team and they provided a patch that can be used until it is properly fixed in the SDK. Here's the note they shared : https://share.note.sx/qfs9kx63#QDsLN9Ji1dVbwTlf34GKYo2XydEnfkMgrnrxyTxLBLE.

If the note ever becomes unavailable, let the team know and they will help you.

Thanks again to the OneSignal team and their help :)

**Comment by fadi-george** [comment]
Hey @fannytavart what RN version & react-native-onesignal version were you using when you were getting this issue? I assume you are using RN New Architecture. I didn't think this error `OneSignalFramework/OneSignalFramework.h` would be related to it but I would like to try and replicate.

Also we released this [version](https://github.com/OneSignal/react-native-onesignal/releases/tag/5.4.3) regarding the codegen issue.

**Comment by wdifruscio** [comment]
Hi, want to chime in here and just say after updating to expo 55 and react native 83.6, my extension is now not working properly.

Usually i just do a clean prebuild and fix some local errors but now I can't seem to get it to link the files properly with this plugin. 

I will try and move to the top as other commenters suggested

**Comment by fadi-george** [comment]
@wdifruscio are you still experiencing issues or did you resolve it?

**Comment by fadi-george** [comment]
Gonna close this for now, if someone is experiencing the same issue then they can file a new issue report hopefully with clear logs and configuration.
